### PR TITLE
Fix TensorRT-LLM

### PR DESCRIPTION
**Pull request by mc-nv**

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -1708,6 +1708,30 @@ def tensorrtllm_prebuild(cmake_script):
     # Export the TRT_ROOT environment variable
     cmake_script.cmd("export TRT_ROOT=/usr/local/tensorrt")
     cmake_script.cmd("export ARCH=$(uname -m)")
+    cmake_script.cmd(
+        'export LD_LIBRARY_PATH="/usr/local/cuda/compat/lib.real:${LD_LIBRARY_PATH}"'
```
Comment (Tabrizian) [inline build.py:1712]: Why is this line required?

**Review by Tabrizian — APPROVED** [review]
@krishung5 might be more familiar with these changes.

File: build.py (hunk 1 above)
Comment (mc-nv) [reply build.py:1712]: Because `libcuda.so.1` can't be resolved till `--runtime=nvidia` is passed. 
It's a workaround that allows us refer to the links in build container, can be deprecated in future. 

You can create task and assign to me.

File: build.py [hunk 2]
```diff
@@ -1102,17 +1102,17 @@ def create_dockerfile_linux(
 RUN python3 -m pip install --upgrade pip \\
       && pip3 install transformers
 
-# Uninstall unused nvidia packages
-RUN if pip freeze | grep -q "nvidia.*"; then \\
-        pip freeze | grep "nvidia.*" | xargs pip uninstall -y; \\
-    fi
-RUN pip cache purge
```
Comment (krishung5) [inline build.py:1109]: I was  wondering why do we need to remove this section? I think this is just to reduce some bytes for the container size. Do we see issues if remove those libraries?

File: build.py [hunk 3]
```diff
@@ -1102,17 +1102,17 @@ def create_dockerfile_linux(
...
-
 # Drop the static libs
 RUN ARCH="$(uname -i)" \\
       && rm -f ${TRT_ROOT}/targets/${ARCH}-linux-gnu/lib/libnvinfer*.a \\
           ${TRT_ROOT}/targets/${ARCH}-linux-gnu/lib/libnvonnxparser_*.a
 
+# Install TensorRT-LLM
+RUN python3 -m pip install /opt/tritonserver/backends/tensorrtllm/tensorrt_llm-*.whl -U --pre --extra-index-url https://pypi.nvidia.com \\
```
Comment (krishung5) [inline build.py:1111]: I see that we are pip installing TRT-LLM in the `Dockerfile.triton.trt_llm_backend`. Do we still need to pip install TRT-LLM again here?

File: build.py (hunk 2 above)
Comment (mc-nv) [reply build.py:1109]: Because it eventually crash the CI

**Review by krishung5 — APPROVED** [review]
